services/adgs/rs_server_adgs/api/adgs_search.py

```diff
@@ -267,11 +268,16 @@ async def get_adgs_collection_specific_item(
 
     # Search all the collection items then search manually for the right one.
     # TODO: allow the search function to take the item ID instead.
-    items = await request.app.state.pgstac_client.item_collection(collection_id, request)
     try:
-        return next(item for item in items.get("features", {}) if item.get("id") == item_id)
-    except StopIteration as exc:
-        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=f"AUXIP item {item_id!r} not found.") from exc
+        item = await request.app.state.pgstac_client.get_item(item_id, collection_id, request)
+    except HTTPException as http_exc:  # validation error, just forward it
```

Comment:
In fact here you don't need the argument, you can just write
```python
except HTTPException:
    raise
```

See: https://stackoverflow.com/a/68560890